src/main/java/com/ldtteam/blockout/Parsers.java

```diff
@@ -0,0 +1,180 @@
+package com.ldtteam.blockout;
+
+import com.ldtteam.structurize.util.LanguageHandler;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.text.IFormattableTextComponent;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraft.util.text.TextFormatting;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.ldtteam.blockout.Log.getLogger;
+
+public final class Parsers
+{
+    private Parsers() { /* prevent construction */ }
+
+    public static final Pattern PERCENTAGE_PATTERN  = Pattern.compile("([-+]?\\d+)(%|px)?", Pattern.CASE_INSENSITIVE);
+    public static final Pattern RGBA_PATTERN        = Pattern.compile("rgba?\\(\\s*(\\d+)\\s*,\\s*(\\d+)\\s*,\\s*(\\d+)\\s*(?:,\\s*([01]\\.\\d+)\\s*)?\\)", Pattern.CASE_INSENSITIVE);
+    public static final Pattern HEXADECIMAL_PATTERN = Pattern.compile("#([0-9A-F]{6,8})", Pattern.CASE_INSENSITIVE);
+
+    // Primitives
+    public static Function<String, Boolean> BOOLEAN = v -> v == null || !v.isEmpty() && !v.equals("disabled") && Boolean.parseBoolean(v);
+    public static Function<String, Integer> INT    = Integer::parseInt;
+    public static Function<String, Float>   FLOAT  = Float::parseFloat;
+    public static Function<String, Double>  DOUBLE = Double::parseDouble;
+
+    public static String NO_TRANSLATION = TextFormatting.OBFUSCATED + "whoops!";
+
+    /** Parses a resource location, include shorthand tricks */
+    public static Function<String, ResourceLocation> RESOURCE = ResourceLocation::new;
+
+    /** Parses a potentially translatable portion of text as a component */
+    @NotNull
+    public static Function<String, IFormattableTextComponent> TEXT = v -> {
+        String result = v == null ? "" : v;
+        Matcher m = Pattern.compile("\\$[({](\\S+)[})]").matcher(result);
+
+        while (m.find())
+        {
+            String translated = LanguageHandler.translateKey(m.group(1));
+            if (translated.equals(m.group(1)))
+            {
+                translated = NO_TRANSLATION;
```

Comment:
Nope, just leave the text as it is, would be hard to debug what is the actual missing key

Reply:
We had 
![image](https://user-images.githubusercontent.com/17338378/105475035-38f3f680-5c9f-11eb-8c79-967bbd8c2aa9.png)